### DISCUSSION TITLE
Wrong generated type for localized text fields

### DISCUSSION THREAD

**Discussion by qbeauperin** [payloadcms/payload#3240] [Feature Requests & Ideas]
### Link to reproduction

Payload > 1.15.0

### To Reproduce

- Create localized text fields in your config
- Run `generate:types`

### Describe the Bug

Take the following config:
```typescript
export const Documents: CollectionConfig = {
    slug: 'documents',
    fields: [
        {
            name: 'title',
            type: 'text',
            required: true,
            localized: true,
        },
        ...
    ]
}
```

It is currently generating the following types:
```typescript
export interface Document {
  id: string;
  title: string;
  ...
}
```

Which does not work when querying with `locale: 'all'`, eg:
```js
{
  id: '64bb323c17c027b62c619846',
  title: { 'en-CA': 'Terms of Service', fr: 'Conditions de service' },
  ...
}
```

I would expect them to be something more like:
```typescript
export interface Document {
  id: string;
  title: string | { [k: string]: string};
  ...
}
```

### Payload Version

1.15.0

**Comment by DanRibbens** [comment]
The only solution I would support on this is to have a `LocaleAllDocument` type that is completely separate from the `Document` type change. The reason we don't want to have `title: string | { [k: string]: string};` is because it will be a breaking type change for a lot of frontends that will need to do some extra type narrowing in a ton of places.

I'm converting this to a discussion as it isn't simply a bug. If anyone can think of another solution, we have a similar problem around the `depth` parameter causing different document structures to be returned than what the types include.

Thanks @qbeauperin!

**Reply by qbeauperin** [reply]
Ah, interesting... 
I'm very much of a Typescript newbie so I'm not sure how much I'll be able to bring to the discussion, but excited to see what others think! 
(a separate type definitely sounds better than nothing though 👍 )
